### ISSUE TITLE
Rep.update returning ok with if: :exists while no record exists and Cassandrax returns false applied

### ISSUE THREAD

**Issue by erobit** [vintikzzz/cassandra_ecto#4]
BTW great job with this package!

I'm having issues with performing Repo.update(user, if: :exists) and having {:ok, %User{...}} returned even though the record does not exist and no updates are applied in Cassandra.

As you can see in the adapter prepare_result method, false/applied returning :ok
https://github.com/vintikzzz/cassandra_ecto/blob/99c839d59c1c4c044196a9d13ea5e07f1b9f56c6/lib/cassandra_ecto/adapter.ex#L294

It's also not clear from the update function signature how to return an :error
https://github.com/elixir-ecto/ecto/blob/v2.1.3/lib/ecto/adapter.ex#L172

Basically the changeset values are being applied instead of an :error being returned. Do you have some insight?

**Comment by vintikzzz** [comment]
Hi, it is quite tricky situation right here. According to https://github.com/elixir-ecto/ecto/blob/v2.1.3/lib/ecto/adapter.ex#L172 it is possible to return `{:invalid, constraints}`, but while doing `Repo.update(user, if: :exists)` we don't really hit any constraint.

I can suggest only to use `Repo.update!(user, if: :exists)` instead for right now.

Can you propose better solution?
